lib/Service/ColumnService.php

```diff
@@ -159,84 +171,125 @@ public function create(
 	/**
 	 * @noinspection PhpUndefinedMethodInspection
 	 * @noinspection DuplicatedCode
-	 * @param int $id
-	 * @param int $tableId
-	 * @param string $userId
-	 * @param string $title
-	 * @param string $type
-	 * @param string $subtype
-	 * @param string $numberPrefix
-	 * @param string $numberSuffix
+	 * @param int $columnId
+	 * @param int|null $tableId
+	 * @param string|null $userId
+	 * @param string|null $type
+	 * @param string|null $subtype
+	 * @param string|null $title
 	 * @param bool $mandatory
-	 * @param string $description
-	 * @param string $textDefault
-	 * @param string $textAllowedPattern
+	 * @param string|null $description
+	 * @param int|null $orderWeight
+	 * @param string|null $textDefault
+	 * @param string|null $textAllowedPattern
 	 * @param int|null $textMaxLength
+	 * @param string|null $numberPrefix
+	 * @param string|null $numberSuffix
 	 * @param float|null $numberDefault
 	 * @param float|null $numberMin
 	 * @param float|null $numberMax
 	 * @param int|null $numberDecimals
-	 * @param string $selectionOptions
-	 * @param string $selectionDefault
-	 * @param int $orderWeight
-	 * @param string $datetimeDefault
+	 * @param string|null $selectionOptions
+	 * @param string|null $selectionDefault
+	 * @param string|null $datetimeDefault
 	 * @return Column
 	 * @throws InternalError
 	 */
 	public function update(
-		int $id,
-		int $tableId,
-		string $userId,
-		string $title,
-		string $type,
-		string $subtype,
-		string $numberPrefix,
-		string $numberSuffix,
-		bool $mandatory,
-		string $description,
-		string $textDefault,
-		string $textAllowedPattern,
+		int $columnId,
+		?int $tableId,
+		?string $userId,
+		?string $type,
+		?string $subtype,
+		?string $title,
+		?bool $mandatory,
+		?string $description,
+		?int $orderWeight,
+
+		?string $textDefault,
+		?string $textAllowedPattern,
 		?int $textMaxLength,
-		?float $numberDefault = null,
-		?float $numberMin = null,
-		?float $numberMax = null,
-		?int $numberDecimals = null,
-		string $selectionOptions = '',
-		string $selectionDefault = '',
-		int $orderWeight = 0,
-		string $datetimeDefault = ''
+
+		?string $numberPrefix,
+		?string $numberSuffix,
+		?float $numberDefault,
+		?float $numberMin,
+		?float $numberMax,
+		?int $numberDecimals,
+
+		?string $selectionOptions,
+		?string $selectionDefault,
+		?string $datetimeDefault
 	):Column {
 		try {
+			$item = $this->mapper->find($columnId);
+
 			// security
-			if (!$this->permissionsService->canUpdateColumnsByTableId($tableId)) {
-				throw new PermissionError('update column id = '.$id.' is not allowed.');
+			if (!$this->permissionsService->canUpdateColumnsByTableId($item->getTableId())) {
```

Comment:
👍 As otherwise there might have been the risk of a security issue with getting the table id if that is just passed from the controllers.